safe_transaction_service/history/services/transaction_service.py

```diff
@@ -109,6 +107,21 @@ def store_txs_in_cache(
 
     # End of cache methods ----------------------------
 
+    def get_count_relevant_txs_for_safe(self, safe_address: ChecksumAddress) -> int:
+        """
+        This method searches multiple tables and count every tx or event for a Safe.
+        It will return the same or higher value if compared to counting ``get_all_tx_identifiers``
+        as that method will group some transactions (for example, 3 ERC20 can be grouped in a ``MultisigTransaction``,
+        so it will be ``1`` element for ``get_all_tx_identifiers`` but ``4`` for this function.
+
+        This query should be pretty fast, and it's meant to be used for invalidating caches.
+
+        :param safe_address:
+        :return: number of relevant txs for a Safe
+        """
+
+        return SafeContract.objects.get_count_relevant_txs_for_safe(safe_address)
```

Comment:
SafeContract is not really used in `get_count_relevant_txs_for_safe`

Reply:
Sure, but queries belong to `models`, so I thought the best place to put the query is on `SafeContract`, as it would be weird to put it under `MultisigTransaction`, `ModuleTransaction` or any of the related classes